site/app/libraries/routers/AccessControl.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace app\libraries\routers;
+
+use Doctrine\Common\Annotations\Annotation;
+
+/**
+ * Annotation class for @AccessControl().
+ *
+ * @Annotation
+ * @Target({"CLASS", "METHOD"})
+ */
```

Comment:
Let's add some documentation/examples of what that might look like, the scope of how you can apply them, do they stack hierarchically (define class level, does defining at method level override it? Can we define a role at class level and permission at method level? We can't and that's fine, but it's good to mention that), etc.

I think we should probably put that documentation here as opposed to `WebRouter::accessCheck` given that IDEs will link here if someone is confused what a particular annotation does. May want to put a comment/link to that particular function though for reference.

Reply:
Several examples have been added. Could you have a look? I will keep it [WIP] until #4030 is good.